### PR TITLE
Cover GetDependenciesTrait

### LINKED ISSUE #140: Increase test coverage
We need to complete test covering, by cover `GetDependenciesTrait`

### PR DESCRIPTION
**Pull request by hans-thomas**
fixes #140
This PR increases the coverage of the `GetDependenciesTrait` class to 100 percent.

### CODE REVIEW

File: tests/SwaggerServiceTest.php [hunk 1]
```diff
@@ -683,6 +685,61 @@ public function testAddDataPostRequestWithObjectParams()
         $service->addData($request, $response);
     }
 
+    public function testAddDataWithNotExistsMethodOnController()
```
Comment (hans-thomas) [inline tests/SwaggerServiceTest.php:688]: This test covers:
https://github.com/RonasIT/laravel-swagger/blob/688fe0b4146b8f9a71abb95e7971c47dcbbc1655/src/Traits/GetDependenciesTrait.php#L16-L18

File: tests/support/Mock/TestController.php [hunk 2]
```diff
@@ -16,7 +16,7 @@ public function testRequestWithAnnotations(TestRequestWithAnnotations $request)
     {
     }
 
-    public function testRequestWithContract(TestContract $contract)
+    public function testRequestWithContract(TestContract $contract, string $param)
```
Comment (hans-thomas) [inline tests/support/Mock/TestController.php:19]: The new string native type covers:
https://github.com/RonasIT/laravel-swagger/blob/688fe0b4146b8f9a71abb95e7971c47dcbbc1655/src/Traits/GetDependenciesTrait.php#L36-L38

File: tests/SwaggerServiceTest.php [hunk 3]
```diff
@@ -683,6 +685,61 @@ public function testAddDataPostRequestWithObjectParams()
...
+        );
+
+        $response = $this->generateResponse('example_success_user_response.json', 200, [
+            'Content-type' => 'application/json',
+        ]);
+
+        $service->addData($request, $response);
+    }
+
+    public function testAddDataWithBindingInterface()
```
Comment (hans-thomas) [inline tests/SwaggerServiceTest.php:715]: Finally, this test is going to cover:
https://github.com/RonasIT/laravel-swagger/blob/688fe0b4146b8f9a71abb95e7971c47dcbbc1655/src/Traits/GetDependenciesTrait.php#L53-L55

File: tests/SwaggerServiceTest.php [hunk 4]
```diff
@@ -683,6 +685,61 @@ public function testAddDataPostRequestWithObjectParams()
...
+    {
+        $this->app->bind(TestContract::class,TestImplementation::class);
```
Comment (DenTray) [inline tests/SwaggerServiceTest.php:717]:
```suggestion
        $this->app->bind(TestContract::class, TestRequest::class);
```

File: tests/SwaggerServiceTest.php [hunk 5]
```diff
@@ -683,6 +685,61 @@ public function testAddDataPostRequestWithObjectParams()
...
+        $this->mockDriverGetTmpData($this->getJsonFixture('tmp_data_get_user_request'));
```
Comment (DenTray) [inline tests/SwaggerServiceTest.php:718]: for this test case we need to:
- bind some test request to the contract
- get this test request class using trait
- parse rules from it
- assert that rules had been parsed and documentation saved.

The current test implementation just call this method but not assert that it work correcting. Please implement last point by using `mockDriverGetEmptyAndSaveTmpData` mock method

File: tests/support/Mock/TestImplementation.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<?php
```
Comment (DenTray) [inline tests/support/Mock/TestImplementation.php:1]: as I described above - we need to test that binded request class returns correct rules and you can use already implemented `TestRequest` class, so this class is useless and should be removed

**Review by DenTray — CHANGES_REQUESTED** [review]

File: tests/SwaggerServiceTest.php (hunk 5 above)
Comment (hans-thomas) [reply tests/SwaggerServiceTest.php]: Please take a look at the new changes. I'm not sure that I get it right.

File: tests/SwaggerServiceTest.php [hunk 7]
```diff
@@ -683,6 +685,61 @@ public function testAddDataPostRequestWithObjectParams()
...
+        $response = $this->generateResponse('example_success_user_response.json', 200, [
+            'Content-type' => 'application/json',
+        ]);
+
+        $service->addData($request, $response);
+    }
+
+    public function testAddDataWithBindingInterface()
+    {
+        $this->app->bind(TestContract::class,TestRequest::class);
```
Comment (DenTray) [inline tests/SwaggerServiceTest.php:717]:
```suggestion
        $this->app->bind(TestContract::class, TestRequest::class);
```

**Review by DenTray — CHANGES_REQUESTED** [review]

**Review by DenTray — APPROVED** [review]

**Pull request merged**